source/posix/thread.c

```diff
@@ -25,6 +25,22 @@
 typedef cpuset_t cpu_set_t;
 #endif
 
+#if !defined(AWS_AFFINITY_METHOD)
+#error "Must provide a method for setting thread affinity"
```

Comment:
This intentionally fails if the method is not specified. It could fall back to NONE, but I considered this more robust as non-cmake build systems can't accidentally build with degraded performance if they are not aware of this.